pkg/scheduler/framework/plugins/interpodaffinity/scoring_test.go

```diff
@@ -929,3 +929,120 @@ func TestPreferredAffinityWithHardPodAffinitySymmetricWeight(t *testing.T) {
 		})
 	}
 }
+
+func TestIgnoreExistingInterPodAffinities(t *testing.T) {
+	podLabelServiceS1 := map[string]string{
+		"service": "S1",
+	}
+	labelRgChina := map[string]string{
+		"region": "China",
+	}
+	labelRgIndia := map[string]string{
+		"region": "India",
+	}
+	labelAzAz1 := map[string]string{
+		"az": "az1",
+	}
+	hardPodAffinity := &v1.Affinity{
+		PodAffinity: &v1.PodAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "service",
+								Operator: metav1.LabelSelectorOpIn,
+								Values:   []string{"S1"},
+							},
+						},
+					},
+					Namespaces: []string{"", "subteam2.team2"},
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "team",
+								Operator: metav1.LabelSelectorOpIn,
+								Values:   []string{"team1"},
+							},
+						},
+					},
+					TopologyKey: "region",
+				},
+			},
+		},
+	}
+	tests := []struct {
+		pod                  *v1.Pod
+		pods                 []*v1.Pod
+		nodes                []*v1.Node
+		interPodAffinityArgs *config.InterPodAffinityArgs
+		expectedList         framework.NodeScoreList
+		name                 string
+	}{
+		{
+			name: "incoming pod does not have inter-pod affinities",
+			pod:  &v1.Pod{ObjectMeta: metav1.ObjectMeta{Labels: podLabelServiceS1}},
+			pods: []*v1.Pod{
+				{Spec: v1.PodSpec{NodeName: "node1"}},
```

Comment:
pls make the two tests exactly the same with the only difference being the value of `IgnoreExistingInterPodAffinities`

Reply:
Also, there is no need to add a new test, just add the cases to the existing one: https://github.com/kubernetes/kubernetes/blob/e74d62ed3010c9244e55a653fecc89122a208ced/pkg/scheduler/framework/plugins/interpodaffinity/scoring_test.go#L371

Reply:
Ok I removed the new test and added two test cases to the existing `TestPreferredAffinity`, which are the same except for the value of `IgnorePreferredTermsOfExistingPods`: 

- `IgnorePreferredTermsOfExistingPods: true` which verifies node scores are all 0
- `IgnorePreferredTermsOfExistingPods: false` which verifies the correct node receives the highest score